### PR TITLE
Add glossary to user guide

### PR DESCRIPTION
**Pull request by bryanhce**
Also updated the activity diagram of markRoomsUnclean in developer guide and added more information on markRoomsUnclean command in user guide.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -182,10 +186,18 @@ Examples:
 
 ### Marking all rooms as unclean: `markRoomsUnclean`
 
-Marks all room clean statuses of all guests as "no".
+This command functions to mark room clean statuses of all guests as "no".
+
+This command is typically used at the end or the start of the day when the hotel manager wishes to 
+update all the room clean statuses to "no" efficiently. This clean slate would allow the hotel 
+managers to keep abreast of the rooms that are cleaned or uncleaned for the day.
```
Comment (jehousoh) [inline docs/UserGuide.md]: This format seems different from the rest of the features, should we reformat the others then?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -236,3 +248,17 @@ _Details coming soon ..._
 | **MarkRoomsUnclean** | `markRoomsUnclean`                                                                                                                                                                             |
```
Comment (jehousoh) [inline docs/UserGuide.md]: Should we add spacing for this MarkRoomsUnclean as it is under "Action"?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -236,3 +248,17 @@ _Details coming soon ..._
...
+| **Alphanumeric** | `Refers to characters that are numbers 0-9 and alphabets A-Z, both uppercase and lowercase.`                                             |
+| **CLI**          | `Stands for Command Line Interface, it refers to a text-based interface used to execute opertaions of the system.`                       |
+| **Guest**        | `A guest staying at the hotel`                                                                                                           |
+| **GUI**          | `Stands for Graphical User Interface, it refers to an interface that allows users to interact with the system through friendly visuals.` |
+| **Hard disk**    | `A data storage device used to store and retrieve data.`                                                                                 |
+| **Index**        | `The number that corresponds to the position of the Guest in the list.`                                                                  |
+| **JSON file**    | `Stands for JavaScript Object Notation, it refers to a file format to store data.`                                                       |
```
Comment (jehousoh) [inline docs/UserGuide.md]: Other things to consider adding. Java, dd/MM/yy, parameter?

**Review by jehousoh — CHANGES_REQUESTED** [review]
Just have some minor comments and considerations, good work!

File: docs/UserGuide.md (hunk 1 above)
Comment (bryanhce) [reply docs/UserGuide.md]: Yes, reformatted some! Will do another around of reformatting when the rest commits their changes.

File: docs/UserGuide.md (hunk 2 above)
Comment (bryanhce) [reply docs/UserGuide.md]: Fixed!

File: docs/UserGuide.md (hunk 3 above)
Comment (bryanhce) [reply docs/UserGuide.md]: Yes, added!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -182,19 +210,31 @@ Examples:
...
+This command is typically used as a hard reset of the entire guest book. Do note that this command 
+is irreversible. Once cleared, all the guest data cannot be retrieved. Please use this command 
+with utmost caution.
 
 Format: `clear`
 
 ### Exiting the program : `exit`
 
-Exits the program.
+This command exits the programme.
```
Comment (jehousoh) [inline docs/UserGuide.md]: Should this be program instead of programme?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -110,13 +129,13 @@ Examples:
...
+This command displays a list of all guests in the guest book.
 
 Format: `list`
 
 ### Editing a guest : `edit`
 
-Edits an existing guest in the guest book.
+This command edits an existing guest in the guest book.
 
 Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [rm/ROOM] [dr/DATE_RANGE] [ng/NUMBER_OF_GUESTS] [rq/REQUEST]`
```
Comment (jehousoh) [inline docs/UserGuide.md]: Should this include rc/?

**Review by jehousoh — CHANGES_REQUESTED** [review]
Some other small changes!

File: docs/UserGuide.md (hunk 5 above)
Comment (bryanhce) [reply docs/UserGuide.md]: Yes, fixed!

**Review by jehousoh — APPROVED** [review]
LGTM!

**Pull request merged**